README.md

```diff
@@ -344,6 +344,28 @@ These settings can be changed at any time.
 * `settings.updates.version-lock`: Controls the version that will be selected when you issue an update request.  Can be locked to a specific version like `v1.0.0`, or `latest` to take the latest available version.  Defaults to `latest`.
 * `settings.updates.ignore-waves`: Updates are rolled out in waves to reduce the impact of issues.  For testing purposes, you can set this to `true` to ignore those waves and update immediately.
 
+#### Network settings
+
+##### Proxy settings
+
+These settings will configure the proxying behavior of the following services:
+* For all variants:
+    * [containerd.service](packages/containerd/containerd.service)
+    * [host-containerd.service](packages/host-ctr/host-containerd.service)
+* For Kubernetes variants:
+    * [kubelet.service](packages/kubernetes-1.18/kubelet.service)
+* For ECS variants:
+    * [docker.service](packages/docker-engine/docker.service)
+    * [ecs.service](packages/ecs-agent/ecs.service)
+
+* `settings.network.http-proxy`: The HTTP proxy server to be used by services listed above.
+* `settings.network.https-proxy`: The HTTPS proxy server to be used by services listed above.
```

Comment:
Do we want these to be separately-configurable at this point?

Reply:
I can't really think of a use-case for configuring `http_proxy` and `https_proxy` differently. But then again there really is no harm in letting them be separately-configurable.

Reply:
It's more overhead for someone who wants to enable a proxy; now they have to configure both.  My preference is to have just one right now and we can always add a second one later if it becomes desirable.

Reply:
Would we still be setting the `HTTPS_PROXY` env var? Setting `HTTPS_PROXY` to the same contents as what's being set for `HTTP_PROXY`?

Reply:
> Would we still be setting the HTTPS_PROXY env var? Setting HTTPS_PROXY to the same contents as what's being set for HTTP_PROXY?

Yep.

Perhaps the hardest question - what to call the setting?  :P

Reply:
Yeah, same value for both.  I'd probably call the setting "https-proxy" to imply that the proxy needs to be able to transparently handle TLS traffic.

Reply:
What if the user wants to use a http proxy that can't handle https traffic though? Wouldn't `https-proxy` be misleading and discourage the user from using the setting?

Reply:
If we're using the same setting for both, the proxy needs to be able to handle HTTPS traffic.  I think the naming should reflect that.

Most of the traffic that will be generated from the software in Bottlerocket (pulling container images, log forwarding, talking to API endpoints) is going to be over HTTPS.  cri-containerd and Docker both require HTTPS when pulling images (without manual overrides that we don't enable in Bottlerocket), and the Kubelet and ECS agent talk to their respective API endpoints over HTTPS as well.